wurst/network/Network.wurst

```diff
@@ -0,0 +1,922 @@
+		sequential, and are received by players in the same order that are sent in.
+
+		It uses the SyncStored* natives of the gamecache to send data to other players
+		from the sender, and then uses SyncSimple to send a 'finish' event, that will
+		only be received by other players after they have also received all sync data.
+
+		The Network class provides 4 independent streams of data:
+			integers,
+			reals,
+			booleans,
+			strings
+
+		each of which can be written/read to using the relevant 
+		methods of the DataStream class.
+
+		Before sending the data, the sender populates the DataStream with the data that
+		they want to send, then Network.start() is called, as well as Network.onFinish(callback)
+		and data is received from the same buffer inside the callback when it has all been received.
+
+		Read SyncSimple docs for a slightly more in-depth overview of this particular
+		peculiarity of WC3.
+
+	Usage:
+		1. Create a new Network object using:
+			// sender is the player to receive data from
+			let network = new Network(sender)
+
+		2. Queue some data for sending from the sender. You can write as much as you need, there is no limit:
+			if localPlayer == sender
+				// keep in mind that before you start sending data,
+				// network.getData() is locked in a WRITE-ONLY MODE
+				val stream = network.getData()
+				stream.writeInteger(...)
+				stream.writeReal(...)
+				stream.writeString(...)
+				...
+
+		3. To start sending the data, simply call:
+			network.start()			
+			// this will start the transmission
+			// keep in mind, that after you call this, you MUST NOT write/read to network.getData()
+			// as it is locked while the data is syncing
+		
+		4. To receive the data, specify the callback and read the data inside it:
+			network.onFinish((stream) -> begin
+				// here, network.getData() is locked in a READ-ONLY MODE and
+				// you can only READ from it
+				var int = stream.readInteger()
+				var real = stream.readReal()
+				var string = stream.readString()
+				...
+			end)
+
+			WARNING: Network is a one-time use class, and is automatically
+			destroyed upon finishing. If you need to send another transmission,
+			create a new one.
+			You are also expected to read all data immediately after receiving it,
+			as the DataStream is destroyed together with Network. 
+
+		5. When writing and reading, keep in mind that you MUST read data from
+		the stream in the same order that it was written, for each independent type.
+		That means that integers have to be written/read in the same order, same for
+		booleans, strings and reals.
+		If you need data of non-fixed length, use a static packet structure
+		with length fields, like so:
+			var field1 = stream.readInteger()
+			var field2 = stream.readInteger()
+			var length = stream.readInteger()
+			for i = 0 to length - 1
+				someArray[i] = stream.readInteger()
+
+		6. Alternatively, your classes can implement Serializable, like so:
```

Comment:
an alternative shouldn't be "the 6th step". Please provide small, concise, correct examples for certain scenarios and separate them accordingly

Reply:
Thanks. Will split into different examples.